File: content/04/code/js/13-replace-hardcoded-sean-instance/features/step_definitions/steps.js
const { Given, When, Then, Before } = require('@cucumber/cucumber')
const { assertThat, is } = require('hamjest')

const { Person, Network } = require('../../src/shouty')

Before(function () {
  this.network = new Network()
  this.people = {}
})

Given('a person named {word}', function (name) {
  this.people[name] = new Person(this.network)
})

When('Sean shouts {string}', function (message) {
  this.people['Sean'].shout(message)
  this.messageFromSean = message
})

Then('Lucy should hear Sean\'s message', function () {
  assertThat(this.people['Lucy'].messagesHeard(), is([this.messageFromSean]))
})

File: content/bdd-with-cucumber/04/code/ruby/13-replace-hardcoded-sean-instance/features/step_definitions/steps.rb
require 'shouty'

Before do
  @network = Shouty::Network.new
  @people  = {}
end

Given('a person named {word}') do |name|
  @people[name] = Shouty::Person.new(@network)
end

When("Sean shouts {string}") do |message|
  @people['Sean'].shout(message)
  @message_from_sean = message
end

Then("Lucy should hear Sean's message") do
  expect(@people['Lucy'].messages_heard).to eq [@message_from_sean]
end
